### ISSUE TITLE
Accessibility: Structured List with selection 

### ISSUE THREAD

**Issue by ligiaomahony** [carbon-design-system/carbon#2567]
## Detailed description

> Describe in detail the issue you're having.

I have a structured list with selection implemented on the application I am working on and at the moment I'm working through a few of the accessibility issues. I did some manual testing and when I got to the list, I can navigate through the list successfully but I cannot select another item in the list using the keyboard only.  

> Is this issue related to a specific component?

Structured List component with selection

> What did you expect to happen? What happened instead? What would you like to see changed?

Item in the list that I have navigated to using the Tab key and which I tried to select using the Space key on the keyboard did not get selected. 

> What browser are you working in?

I am using Chrome and Firefox. Same issue across both vrowsers.

> What version of the Carbon Design System are you using?

![image](https://user-images.githubusercontent.com/44969729/48345762-d8091680-e670-11e8-9787-736fd967f6fc.png)

## Steps to reproduce the issue
1. Use Tab to navigate through list 
2. When I want to select a different item from the list I click the space key. The checkmark should be now be applied to the item I just selected using spacebar but none was applied. 
3. The structured list item does not get selected

* Screenshots or code
![image](https://user-images.githubusercontent.com/44969729/48345486-0c300780-e670-11e8-9f6f-bb49f7f7069e.png)
